### PR TITLE
[CST-1063] Cannot add participant from different school message

### PR DESCRIPTION
**Pull request by lazydevorg**
### Context

- Ticket: CST-1063

### Changes proposed in this pull request
The "cannot add" logic checks whether the existing participant is in the same school of the new participant.
The message shown changes based on whether the above condition is true or not.

### CODE REVIEW

File: app/forms/schools/add_participant_form.rb [hunk 1]
```diff
@@ -358,5 +358,9 @@ def appropriate_body_selected
     rescue ActiveRecord::RecordNotFound
       nil
     end
+
+    def existing_participant_profile_same_school?
+      existing_participant_profile&.school == school_cohort.school
```
Comment (ltello) [inline app/forms/schools/add_participant_form.rb:363]: Surely ParticipantProfile::ECF.school is legacy. 
Shouldn't we use the participant_profile's latest_induction_record's school instead ?

Comment (tonyheadford) [reply app/forms/schools/add_participant_form.rb:363]: Yes I agree, lets use the induction records for this :+1:

File: app/views/schools/add_participants/cannot_add.html.erb [hunk 2]
```diff
@@ -7,7 +7,11 @@
     <span class="govuk-caption-xl"><%= @school.name %></span>
     <h1 class="govuk-heading-xl"> You cannot add <%= add_participant_form.full_name.titleize %></h1>
 
-    <p class="govuk-body">Our records show this person is already registered on an ECF-based training programme at a different school</p>
+    <% if add_participant_form.existing_participant_profile_same_school? %>
+      <p class="govuk-body">Our records show this person is already registered on an ECF-based training programme at this school</p>
+    <% else %>
+      <p class="govuk-body">Our records show this person is already registered on an ECF-based training programme at a different school</p>
+    <% end %>
```
Comment (tonyheadford) [inline app/views/schools/add_participants/cannot_add.html.erb:14]: Could we DRY this up a bit?

File: app/forms/schools/add_participant_form.rb [hunk 3]
```diff
@@ -358,5 +358,9 @@ def appropriate_body_selected
     rescue ActiveRecord::RecordNotFound
       nil
     end
+
+    def existing_participant_profile_same_school?
+      existing_participant_profile&.current_induction_record&.school == school_cohort.school
```
Comment (ltello) [inline app/forms/schools/add_participant_form.rb:363]: I wonder if it'd be better to use the latest induction record no matter its induction_status:
`existing_participant_profile&.induction_records&.latest&.school == school_cohort.school`

**Review by ltello — APPROVED** [review]

**Review by ltello — APPROVED** [review]

**Pull request merged**